### PR TITLE
release 2.4-M1

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -25,6 +25,18 @@ title: Akka Documentation
 
 * Akka API - for [Java](http://doc.akka.io/japi/akka/2.2.5/) and [Scala](http://doc.akka.io/api/akka/2.2.5/)
 
+## Milestone Versions
+ 
+### Akka 2.4-M1 (pre-release of what is going to become Akka 2.4) for Scala 2.11 and 2.12
```
Comment (rkuhn) [inline docs/index.md]: please change to “for Scala 2.11 / 2.12 and Java 8+”, also change for the Release Versions (including Java 6+, obviously)

Comment (ktoso) [inline docs/index.md]: Do we want to say `2.12.M1` during 12s milestone phase?

File: news/_posts/2015-05-22-akka-2.4-M1-released.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+* experimental modules, such as akka-persistence and akka-contrib
+* features, classes, methods that were deprecated in 2.3.x and removed in 2.4.x
+
+The dependency to Netty has been updated from version 3.8.0.Final to 3.10.3.Final. The changes in those versions might not be fully binary compatible, but we believe that it will not be a problem in practice. No changes were needed to the Akka source code for this update. Users of libraries that depend on 3.8.0.Final that break with 3.10.3.Final should be able to manually downgrade the dependency to 3.8.0.Final and Akka will still work with that version.
+
+The dependency to Typesafe Config has been updated from 1.2.1 to 1.3.0 which should be binary compatible for the vast majority users, except for obscure edge cases as its [changelog](https://github.com/typesafehub/config/blob/master/NEWS.md#130-may-8-2015) points out. This change was made in order in order to use new JDK8 specific features in the library as well as to align Akka with Play which is now also depending on 1.3.0.
+
+When migrating a code base to 2.4 please refer to the migration guide in order to profit from some of the improvements.
```
Comment (rkuhn) [inline news/_posts/2015-05-22-akka-2.4-M1-released.md]: please add a link to the migration guide

File: docs/index.md (hunk 1 above)
Comment (patriknw) [inline docs/index.md]: yes, that is probably more correct, 2.12.0-M1 even

File: news/_posts/2015-05-22-akka-2.4-M1-released.md (hunk 2 above)
Comment (patriknw) [inline news/_posts/2015-05-22-akka-2.4-M1-released.md]: good catch, thanks

File: news/_posts/2015-05-22-akka-2.4-M1-released.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+* we dropped support for Scala 2.10, kept 2.11 and added 2.12 (which is at milestone 1 currently as well); this is in keeping with our policy to support the Scala version that is “current” when the first milestone comes out plus any later version that is published during this series’ lifetime
+* we added the experimental [Akka Typed](http://doc.akka.io/docs/akka/2.4-M1/scala/typed.html) module previously codenamed Project Gålbma, a new way of formulating Actor interactions with full type-system support
+* we promoted the [ClusterSingleton](http://doc.akka.io/docs/akka/2.4-M1/scala/cluster-singleton.html), [ClusterClient](http://doc.akka.io/docs/akka/2.4-M1/scala/cluster-client.html), [DistributedPubSub](http://doc.akka.io/docs/akka/2.4-M1/scala/distributed-pub-sub.html) and [ClusterSharding](http://doc.akka.io/docs/akka/2.4-M1/scala/cluster-sharding.html) patterns to fully supported Cluster tools
+* ClusterSharding also learnt a few new tricks (asynchronous shard allocation, reviving entries after migration [thanks to [Dominic Black](https://github.com/DomBlack)], and more flexible use of roles [thanks to Richard Marscher](https://github.com/rmarsch)])
```
Comment (rkuhn) [inline news/_posts/2015-05-22-akka-2.4-M1-released.md]: There is an opening [ missing before Richard Marscher.

File: docs/index.md (hunk 1 above)
Comment (rkuhn) [inline docs/index.md]: agreed

**Comment by rkuhn** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM

**Comment by patriknw** [comment]
I have corrected the misstakes

**Pull request merged**